update

```diff
@@ -59,19 +65,35 @@ do_fixedref()
     )
 }
 
-checkout_ffmpeg=do_gitmaster
-#checkout_fribidi=do_releasetag
-checkout_libass=do_gitmaster
-checkout_mpv=do_gitmaster
+# args: $1: project name, $2: release/master [,$3: non-empty to ignore the config file]
+checkout()
+{
+    local arg="$2"
```

Comment:
`branch` would be more descriptive than arg.

Reply:
It's not always a branch either, because the next commit adds support for a custom commit/tag/branch. Maybe `version` or `checkout` would have been a more appropriate name. But I also didn't want to modify the config file names branch-$project . So yeah, I'll make it `branch`.

> I hate to suggest adding more scripts, but what if someone wants a specific libass version? I think the reason it currently doesn't have such scripts is only to discourage use of its releases which aren't kept up to date. Maybe a use-libass-master, use-libass-custom are in order?

I feel the same. In general, this whole `update` script is just supposed to make it easier for people to pull and possibly checkout a specific revision of the projects. The only "smart" thing it knows to do otherwise is to map "release" to the latest release tag.

All those scripts are just supposed to make it easier to set a value of release/master/custom per project.

How about this: I'll add 3 more libass scripts for symmetry (or just 2 like you suggested, I don't mind at all), and then do a followup to unify all those 8/9 scripts, together with the additional projects options, to one config file which could have comments in it etc?

This single config would be a one-stop-shop for anyone who wants to customize it. Makes sense?

Reply:
That sounds fine, though I think I'd lean towards not having a libass master script, if nothing else than to reduce the sheer number of scripts unless you're suggesting somehow merging them too. There is something to be said about `ls` basically providing "documentation", but there's a point where more scripts are more inundating than helpful.

Reply:
> unless you're suggesting somehow merging them too

That's what I'm suggesting. One config file to replace all those use-foo-bar scripts, together with the options per project config values. All the config in one place, with a default config file which can be copied and modified by users, or some such.

Reply:
Oh, yeah, sure that works. Much simpler. I guess it's not too much to ask to edit a file if you're that particular about your versions.